[PPWM-39] [PPWM-43] [PPWM-16] Various Migration Actor Fixes

ehigham opened the pull request:
The following makes the`WorkspaceMigrationActor` successfully migrate a simple workspace.

RR: https://broadworkbench.atlassian.net/browse/PPWM-39
Fix migration failure while setting up google project IAM. Caused by
using the wrong group as the billing project owners group - I had
assumed that this would be the same group as the billing project owners
policy on the workspace.

RR: https://broadworkbench.atlassian.net/browse/PPWM-43
Fix 403 when trying to create the google project resource in sam with
the workspace as its parent by using sam's resource admin apis to
temporarily add rawls as an owner to the workspace resource.

RR: https://broadworkbench.atlassian.net/browse/PPWM-15
Expose `updateBucketIam` in the `GoogleServicesDAO` (data access
objects contain business logic because of course they do) and call it
from the migration actor.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

ehigham committed:
[PPWM-39] Get Billing Project Owners Group from Billing Project
RR: https://broadworkbench.atlassian.net/browse/PPWM-39
Fix migration failure while setting up google project IAM. Caused by
using the wrong group as the billing project owners group - I had
assumed that this would be the same group as the billing project owners
policy on the workspace.

dvoet commented on core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationActor.scala:
```diff
@@ -439,31 +431,33 @@ object WorkspaceMigrationActor {
-              accessPolicies,
-              billingProjectOwnerPolicyEmail
+              workspacePolicies.map(p => p.policyName -> p.email).toMap,
+              billingProjectOwnerPolicy.email
```
it is more complicated that this: https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L2480

the billingProjectOwnerPolicyEmail comes from the billing project only when there isn't an auth domain

ehigham replied:
This comes from https://github.com/broadinstitute/rawls/blob/39cbdba87097d0be3b798c2dcb2f1f2ba4b748e8/core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala#L2445, and that email is used by `setupGoogleProjectIam` in the same function, before the code you linked. Let me know what to do because I'm finding it really hard to unpack that function.

ehigham replied:
Done when updating bucket Iam

ehigham committed:
[PPWM-43] Add Google Project Resource to Workspace as Sam Admin
RR: https://broadworkbench.atlassian.net/browse/PPWM-43
Fix 403 when trying to create the google project resource in sam with
the workspace as its parent by using sam's resource admin apis to
temporarily add rawls as an owner to the workspace resource.

ehigham committed:
[PPWM-15] Update Bucket IAM At End of Migration
RR: https://broadworkbench.atlassian.net/browse/PPWM-15
Expose `updateBucketIam` in the `GoogleServicesDAO` (data access
objects contain business logic because of course they do) and call it
from the migration actor.

ehigham committed:
use billing project owner's group when workspace not in auth domain

ehigham committed:
fix test build failure

ehigham committed:
Merge branch 'develop' into ehigham--ppwm-39-fix-update-google-project-iam-failure

ehigham committed:
fix comment

ehigham committed:
Merge branch 'develop' into ehigham--ppwm-39-fix-update-google-project-iam-failure
